atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2576,6 +2583,37 @@ private Multimap<TableReference, Cell> getCellsToScrubByTable(State expectedStat
         return tableRefToCells;
     }
 
+    @Override
+    public long getAgeMillis() {
+        return System.currentTimeMillis() - timeCreated;
+    }
+
+    @Override
+    public TransactionReadInfo getReadInfo() {
+        return keyValueService.getOverallReadInfo();
+    }
+
+    @Override
+    public void reportExpectationsCollectedData() {
+        if (!List.of(State.COMMITTED, State.ABORTED, State.FAILED).contains(state.get())) {
+            log.error(
+                    "reportExpectationsCollectedData is called on an in-progress transaction",
```

Comment:
You can probably reuse some of the logic from `ensureStillRunning()` - it isn't the same, but there should be similar bits you can use :)